src/runtimes/go/index.ts

```diff
@@ -139,13 +140,18 @@ const zipFunction: ZipFunction = async function ({
     const zipPath = `${destPath}.zip`
     const zipOptions = {
       destPath: zipPath,
-      filename: basename(destPath),
+      filename: featureFlags.zisi_golang_use_al2 ? 'bootstrap' : basename(destPath),
       runtime,
     }
 
     await zipBinary({ ...zipOptions, srcPath: binary.path, stat: binary.stat })
 
-    return { config, path: zipPath, entryFilename: zipOptions.filename }
+    return {
+      config,
+      path: zipPath,
+      entryFilename: zipOptions.filename,
+      runtimeVersion: featureFlags.zisi_golang_use_al2 ? 'provided.al2' : undefined,
```

Comment:
Just for consistency with what we do in Node.js, it might be nice to always return the runtime version, even if it's the old one.

Not a big deal, though.

```suggestion
      runtimeVersion: featureFlags.zisi_golang_use_al2 ? 'provided.al2' : 'go1.x',
```

Reply:
Thought about that as well, but I didn't want to introduce changes when the flag is off. `undefined` should do that.